caput/fileformats.py

```diff
@@ -88,7 +88,9 @@ def compression_kwargs(compression=None, compression_opts=None, compressor=None)
         super(HDF5, HDF5).compression_kwargs(compression, compression_opts, compressor)
         if compressor:
             raise NotImplementedError
-        if compression in ("bitshuffle", H5FILTER, str(H5FILTER)):
+        if (compression == "bitshuffle") or (
+            (H5FILTER is not None) and compression in (H5FILTER, str(H5FILTER))
+        ):
```

Comment:
Given that whenever `H5FILTER` is set to `None` (up on Line 27), then `compression_enabled` is also set to `False`, and recognising that the only value of `compression` that can cause `ValueError` to get raised is the `"bitshuffle"` literal, it might be more legible to pull the `raise ValueError` check that happens immediately after this `if` out of the `if`-block and turn this into something like:
```suggestion
        if compression == "bitshuffle" and not compression_enabled:
            raise ValueError(
                "Install with 'compression' extra_require to use bitshuffle/numcodecs compression filters."
            )

        if compression_enabled and compression in ("bitshuffle", H5FILTER, str(H5FILTER)):
```
(Also need to delete the next four lines containing the `if not compression_enabled` block.)

Reply:
Thanks @ketiltrout!  I've implemented this suggestion.